### PR TITLE
Minimal support for SavedModel in WarmStartSettings

### PR DESCRIPTION
**Pull request by dwyatte**
This PR adds support for using a SavedModel with `WarmStartSettings` by passing `[saved_model_export_dir]/variables` to `WarmStartSettings.ckpt_to_initialize_from` as discussed in https://github.com/tensorflow/tensorflow/issues/25400. 

It may be nicer to just change `WarmStartSettings` to accept the entire SavedModel just like  `Estimator.warm_start_from`, but this is the more minimal change and may only be for v1 compatibility since tests are marked `run_v1_only`.

### CODE REVIEW

**Comment by gbaned** [comment]
@k-w-w  Can you please review this PR ? Thanks!

File: tensorflow_estimator/python/estimator/estimator.py [hunk 1]
```diff
@@ -2355,23 +2355,36 @@ def _get_default_warm_start_settings(warm_start_from):
...
+        os.path.join(
+            saved_model_utils.get_variables_dir(path),
+            tf.compat.as_text('variables.index')))
+
+  def _is_saved_model_variables_dir(path):
+    """Infer that path is a SavedModel variables dir if ends with 'variables'
+    and parent directory is a SavedModel"""
+    path = tf.compat.as_text(os.path.normpath(path))
+    return path.endswith('variables') and _is_saved_model(os.path.dirname(path))
+
```
Comment (k-w-w) [inline tensorflow_estimator/python/estimator/estimator.py:2370]: Move nested functions outside if they are not using local variables

File: tensorflow_estimator/python/estimator/estimator.py [hunk 2]
```diff
@@ -2355,23 +2355,36 @@ def _get_default_warm_start_settings(warm_start_from):
...
-              warm_start_from))
+      warm_start_from = saved_model_utils.get_variables_path(warm_start_from)
     return WarmStartSettings(ckpt_to_initialize_from=warm_start_from)
   elif isinstance(warm_start_from, WarmStartSettings):
+    if _is_saved_model_variables_dir(warm_start_from.ckpt_to_initialize_from):
+      # replace WarmStartSettings to point to variables path
+      tf.compat.v1.logging.info('Warm-starting from a SavedModel')
+      variables_path = os.path.join(
+          tf.compat.as_text(warm_start_from.ckpt_to_initialize_from),
+          'variables')
```
Comment (k-w-w) [inline tensorflow_estimator/python/estimator/estimator.py:2385]: Change to this constant: `tf.saved_model.VARIABLES_FILENAME` (also wrap with compat.as_text)

**Review by k-w-w — COMMENTED** [review]
Thanks!

**Comment by gbaned** [comment]
@k-w-w Can you please review this PR ? Thanks!

**Comment by gbaned** [comment]
@k-w-w Can you please review this PR ? Thanks!

**Review by mihaimaruseac — APPROVED** [review]

**Comment by mihaimaruseac** [comment]
Closing stale PR. Note that Estimator is no longer active, currently it's only updated as needed to fully deprecate from TF.

**Pull request closed without merging**